nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py

```diff
@@ -168,7 +180,12 @@ def get_metadata(self):
         session_path = Path(self.source_data["folder_path"])
         session_id = session_path.stem
         metadata = NeuroscopeRecordingInterface.get_ecephys_metadata(
-            xml_file_path=str((session_path / f"{session_id}.xml").absolute())
+            xml_file_path=str(session_path / f"{session_id}.xml"),
+            recording=NumpyRecordingExtractor(
+                timeseries=np.array([]), sampling_frequency=self.sorting_extractor.get_sampling_frequency
+            )
+            # TODO, with new recording channel property format for heavy metadata, need to call add_electrodes
+            # on the data contained in this dummy RecordingExtractor
```

Comment:
I don't like this workflow. A `get` method should not make internal state changes